src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -967,6 +977,35 @@ def update_vm(cmd, resource_group_name, vm_name, os_disk=None, disk_caching=None
     return sdk_no_wait(no_wait, _compute_client_factory(cmd.cli_ctx).virtual_machines.create_or_update,
                        resource_group_name, vm_name, **kwargs)
 
+
+def _prepare_workspace(cmd, resource_group_name, workspace):
+    from msrestazure.tools import is_valid_resource_id
+    from ._client_factory import cf_log_analytics
+    from azure.cli.core.commands.client_factory import get_subscription_id
+
+    workspace_id = None
+    if not is_valid_resource_id(workspace):
+        workspace_name = workspace
+        subscription_id = get_subscription_id(cmd.cli_ctx)
+        log_client = cf_log_analytics(cmd.cli_ctx, subscription_id)
+        workspace_result = None
+        try:
+            workspace_result = log_client.get(resource_group_name, workspace_name)
+        except:  # pylint: disable=bare-except
+            from azure.mgmt.loganalytics.models import Workspace, Sku
+            sku = Sku(name="pergb2018")
```

Comment:
```
class SkuNameEnum(str, Enum):

    free = "Free"
    standard = "Standard"
    premium = "Premium"
    unlimited = "Unlimited"
    per_node = "PerNode"
    per_gb2018 = "PerGB2018"
    standalone = "Standalone"
```
I find a class in azure/mgmt/loganalytics/models/log_analytics_management_client_enums.py.
Avoid using magic numbers and string literals in your code.

Reply:
I will delete the magic string. Actually, server can only support pergb2018. Other SKU are disabled by server.